File: arrays-1d/continuousSumQuery/optimised.cpp
vector<int> Solution::solve(int A, vector<vector<int> > &B) {
    vector<int> arrA(A,0);

    for(int i=0;i<B.size();i++){
        int L = B[i][0];
        int R = B[i][1];
        int X = B[i][2];

        arrA[L-1] += X;
        if(R<arrA.size()) {
            arrA[R] -= X;
        }
    }

    for(int i=1;i<arrA.size();i++){
        arrA[i] += arrA[i-1];
    }

    return arrA;
}

File: array-techniques/rotatedArray/optimised.py
MOD_VAL = 1000000009

class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return a list of integers
    def solve(self, A, B):
        A_rotated = [0]*len(A)

        for i in range(len(A)):
            A_rotated[((i % MOD_VAL + B % MOD_VAL) % MOD_VAL) % len(A)] = A[i]
        
        return A_rotated
